st2common/st2common/models/system/paramiko_script_action.py

```diff
@@ -49,7 +50,11 @@ def _format_command(self):
                 else:
                     command = quote_unix('cd %s && %s' % (cwd, script_path))
 
-            command = 'sudo -E -- bash -c %s' % (command)
+            sudo_arguments = ' '.join(self._get_common_sudo_arguments())
+            command = 'sudo %s -- bash -c %s' % (sudo_arguments, command)
+
+            if self.sudo_password:
+                command = 'echo -e %s | %s' % (quote_unix('%s\n' % (self.sudo_password)), command)
```

Comment:
What about command history for bash? Would this not show up in cleartext in `~/.bash_history` by default?

Reply:
That's correctly - sadly not much we can do it besides documenting this (it affects remote runner, but not local one).

Well, we could actually try to set some environment variables and try to disable bash history, but this it not a bullet-proof solution anyway.

It's similar with other actions which pass sensitive / secret stuff as command line arguments. We already have best practice docs which says sensitive data should be retrieved inside an action to reduce surface area and prevent exposure.

I will also document that limitation when using sudo password.